### ISSUE TITLE
Analytics to get information on what versions are still active

### ISSUE THREAD

**Issue by nitrocode** [asdf-vm/asdf#1283]
### Is your feature request related to a problem? Please describe

I'd like to know what versions of tools are still active such as terraform to understand what my custom tools should be able to support.

### Describe the proposed solution

Ability to have the asdf provide analytics like homebrew with a way of opting in (or opting out like with homebrew)

It could provide a nice UI like homebrew does for its formulas

https://formulae.brew.sh/formula/terraform

### Describe similar `asdf` features and why they are not sufficient

None that I know of

### Describe other workarounds you've considered

None that I know of

**Comment by jthegedus** [comment]
Can you elaborate?

>I'd like to know what versions of tools are still active such as terraform

What do you mean by "active". `asdf list all terraform` should tell you the versions supported by your `terraform` plugin.

`asdf plugin list --urls` will show you the location of the plugin so you can read it's documentation, code etc.

---

asdf, is not like Homebrew in that it does not control the entire end-to-end solution, it relies on plugins adhering to a simple API.

You can install plugins from anywhere. The asdf core maintain 4 plugins of the ~500 tracked in our central plugin repo. We do not control most of the plugins. This allows you to easily write and use plugins for any tool without relying on anyone else or availability of any external service. This is because the central repo is not required, it simply provides a shortcut and discovery of plugins. We're looking into replacing it with other means of finding plugins without the burden the central plugin repo creates.

Regarding analytics, I believe the asdf core team are aligned in that we are against any form of tracking within the tool. Again, because each plugin is usually a GitHub or GitLab repository you can view metrics about the plugins at the repository source. You can see commit history, all code and even number or repo clones etc.

I personally do not see the value of trying to emulate the Homebrew UI you share for asdf. The required data depends on the plugin, and we do not control the plugins. I see this as a positive.

**Comment by Stratus3D** [comment]
> Regarding analytics, I believe the asdf core team are aligned in that we are against any form of tracking within the tool. Again, because each plugin is usually a GitHub or GitLab repository you can view metrics about the plugins at the repository source. You can see commit history, all code and even number or repo clones etc.
>
> I personally do not see the value of trying to emulate the Homebrew UI you share for asdf. The required data depends on the plugin, and we do not control the plugins. I see this as a positive.

I agree. I think it's important to our users not to be leaking their information. Even if it's just aggregate usage stats that gets annoymized. 

That said I understand the need for local metrics. It might be nice to know things like which versions you haven't used at all in the last month, and how often you use different versions of a tool. I think however this can be tracked under https://github.com/asdf-vm/asdf/issues/1717 so I'm going to close this issue.
